ee/billing/quota_limiting.py

```diff
@@ -0,0 +1,31 @@
+from datetime import timedelta
+from enum import Enum
+from typing import List, Mapping
+
+from django.utils import timezone
+
+from posthog.cache_utils import cache_for
+from posthog.redis import get_client
+
+RATE_LIMITER_CACHE_KEY = "@posthog/quota-limits/"
+
+
+class QuotaResource(Enum):
+    EVENTS = "events"
+    RECORDINGS = "recordings"
+
+
+def replace_limited_teams(resource: QuotaResource, tokens: Mapping[str, int]) -> None:
+    pipe = get_client().pipeline()
+    pipe.delete(f"{RATE_LIMITER_CACHE_KEY}{resource.value}")
+    if tokens:
+        pipe.zadd(f"{RATE_LIMITER_CACHE_KEY}{resource.value}", tokens)  # type: ignore # (zadd takes a Mapping[str, int] but the derived Union type is wrong)
+    pipe.execute()
+
+
+@cache_for(timedelta(seconds=30), background_refresh=True)
+def list_limited_teams(resource: QuotaResource) -> List[str]:
```

Comment:
nit: My initial confusion with teams may have been supported by this function name. Given that we are returning tokens/api keys, why not `list_limited_api_keys` / `list_limited_tokens`?

Not blocking at all.